src/applications/model/three-lognormal-random-variable.h

```diff
@@ -0,0 +1,99 @@
+/* -*- Mode:C++; c-file-style:"gnu"; indent-tabs-mode:nil; -*- */
+/*
+ * Copyright (c) 2020, University of Padova, Department of Information
+ * Engineering, SIGNET Lab.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation;
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ * Authors: Salman Mohebi <s.mohebi22@gmail.com>
+ *
+ */
+
+#ifndef THREE_LOGNORMAL_RANDOM_VARIABLE_H
+#define THREE_LOGNORMAL_RANDOM_VARIABLE_H
+
+#include "ns3/random-variable-stream.h"
+
+namespace ns3 {
+
+/**
+ * \ingroup randomvariable
+ * \brief The three-parameters log-normal distribution Random Number Generator
+ * (RNG) that allows stream numbers to be set deterministically.
+ *
+ * The three-parameter log-normal distribution is simply the usual two-parameter
+ * log-normal distribution with a location shift.
+ * Let X be a random variable with a three-parameter log-normal distribution
+ * with parameters meanlog=mu, sdlog=sigma, and threshold=threshold
+ * Then the random variable Y=X−γ has a log-normal distribution with parameters
+ * meanlog=mu and sdlog=sigma
+ */
+class ThreeLogNormalRandomVariable : public LogNormalRandomVariable
+{
+public:
+  /**
+   * \brief Register this type.
+   * \return The object TypeId.
+   */
+  static TypeId GetTypeId (void);
+
+  /**
+   * \brief Creates a log-normal distribution RNG with the default
+   * values for mu, sigma and threshold.
+   */
+  ThreeLogNormalRandomVariable ();
+
+  /**
+   * \brief Returns the threshold value for the three-parameters log-normal distribution returned by this RNG stream.
+   * \return The threshold value for the three-parameters log-normal distribution returned by this RNG stream.
+   */
+  double GetThreshold (void) const;
+
+  /**
+   * \brief Returns a random double from a log-normal distribution with the specified mu and sigma.
+   * \param [in] threshold The threshold value for the three-parameters log-normal distribution.
+   * \return A floating point random value.
+   */
+  double GetValue (double threshold);
```

Comment:
Meaningless. Maybe you wanted to add
```suggestion
  double GetValue (double mu, double sigma, double threshold);
```

Reply:
I'm calling this method from the `virtual double GetValue (void)` method.
 
The `mu` and `sigma` are the private attributes of the parent class (`LogNormalRandomVariabl`) and if I define the function with three parameters, I need to call it from `virtual double GetValue (void)` like this:
`GetValue(LogNormalRandomVariable::GetMu(),LogNormalRandomVariable::GetSigma(),m_threshold);`

Reply:
I can either change the `LogNormalRandomVariable` class and make the `mu` and `sigma` protected, add these attributes to ThreeLogNOrmalRandomVariable .

Reply:
take a look at the implementation of the equivalent method in the base class. Its use is to ignore the member variables and to instead compute a lognormal r.v. with the given parameters. This should be extended to the derived class as well, no need to access `LogNormalRandomVariable`'s private member variables

Reply:
You right I got the point. This method should have three parameters.

But the problem I said above still remains:
The `GetValue (void)` method is used to call this method (`GetValue(mu, sigma, threshold)`) with the default member variables. 
But I don't have access to `m_mu` and `m_sigma` from a derived class. 
Now the question is: is it ok to call this function like this:
`GetValue(LogNormalRandomVariable::GetMu(),LogNormalRandomVariable::GetSigma(),m_threshold); `?

Reply:
sorry i overlooked that part.
You can call it as `GetValue(GetMu(), GetSigma(), m_threshold);`

Reply:
Yeah, you right. I have access to these methods in the child class as well. :-)
Thanks